cmd/verify.go

```diff
@@ -137,15 +137,30 @@ func runVerify(ctx context.Context, vo options.VerifyOptions, verifiers ...crypt
 		return fmt.Errorf("failed to verify policy: %w", err)
 	}
 
-	log.Info("Verification succeeded")
-	log.Info("Evidence:")
-	num := 0
-	for _, stepEvidence := range verifiedEvidence {
-		for _, e := range stepEvidence {
-			log.Info(fmt.Sprintf("%d: %s", num, e.Reference))
-			num++
+	if result {
+		log.Info("Verification succeeded")
+		log.Info("Evidence:")
+		num := 0
+		for step, result := range verifiedEvidence {
+			log.Info("Step: ", step)
+			for _, p := range result.Passed {
+				log.Info(fmt.Sprintf("%d: %s", num, p.Reference))
+				num++
+			}
+		}
+	} else {
+		log.Error("Verification failed")
+		log.Error("Evidence:")
+		for step, result := range verifiedEvidence {
+			log.Error("Step: ", step)
+			for _, p := range result.Rejected {
+				if p.Collection.Collection.Name != "" {
+					log.Errorf("Collection Rejected: %s, Reason: %s ", p.Collection.Collection.Name, p.Reason)
+				} else {
+					log.Errorf("Verification Failure: Reason: %s", p.Reason)
+				}
+			}
```

Comment:
We miss something here that says return final error so because we had a failure. Right now we do not have a exit code 1. My test pipeline did not fail although the policy verify did not succeed.

Reply:
Ah this is my bad! I'll sort this once I've had lunch 🥙 thank you for reviewing!

Reply:
Another thing that I'd love feedback on here is that with this change, policy verification failure *does not* bring about an `err != nil` response.

This is fine in my opinion as far as just witness CLI is concerned. However, I wonder if those using the function via `go-witness` would overlook this / find it misleading.

The thoughts of others would be very helpful here 😊

Reply:
I think its good this way. A tool can desde how to respond to the verification report details. You cant capture that well in an error only. Or you need to return error and the results. But that is counter intuïtieve from an api perspective in go.

Maybe the result object could have a Failed method that returns boolean that you can use to check if the verification Succeded. So some convenience methods.

An actual error should happen on an invalid policy file for example. Or corrupt keyid. Thing like that.

Reply:
Agreed @matglas. Could you confirm that now you're seeing exit 1 now? I've changed things so it just passes the error back to cobra which should handle the exit 1.

Reply:
I see one challenge here. If there are multiple results and reasons for failing you will only see the first one. You will not be able to address multiple issues at once when you discover it. This will pose a problem later on. I believe you need to log the error(s) to the output but fail with one final error.

Reply:
Thanks for catching this, this was my bad! Went to fast and wasn't paying attention 😄